hoot-core/src/main/cpp/hoot/core/elements/ConstElementVisitor.h

```diff
@@ -51,9 +51,18 @@ class ConstElementVisitor
   virtual void visit(const ConstElementPtr& e) = 0;
 
   /**
-    Returns a string representation of the visitor
+    Returns a user friendly message indicating the visitor has begun its work
+
+    e.g. "Removing duplicate ways..."
+    */
+  virtual QString getInitStatusMessage() { return ""; }
+
+  /**
+    Returns a user friendly message indicating a summary of the work performed by the visitor
+
+    e.g. "Removed 20 duplicate ways"
     */
-  virtual QString toString() { return ""; }
+  virtual QString getCompletedStatusMessage() { return ""; }
 
```

Comment:
Making these optional on vis/ops both due to the sheer amount of work to add them all at once and the fact that not every vis/op really needs to display the message...but probably most of them eventually will.